File: GeneratedEntityReaderClasses/Components/TorchComponent.cs
namespace GeneratedNoitaClasses;
public class TorchComponent : NoitaComponentBase,  iNoitaType<TorchComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt probability_of_ignition_attempt { get; set; } = new();
	public NoitaFloat suffocation_check_offset_y { get; set; } = new();
	public NoitaInt frames_suffocated_to_extinguish { get; set; } = new();
	public NoitaBool extinguishable { get; set; } = new();
	public NoitaFloat fire_audio_weight { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		probability_of_ignition_attempt.Read(s);
		suffocation_check_offset_y.Read(s);
		frames_suffocated_to_extinguish.Read(s);
		extinguishable.Read(s);
		fire_audio_weight.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		probability_of_ignition_attempt.Write(s);
		suffocation_check_offset_y.Write(s);
		frames_suffocated_to_extinguish.Write(s);
		extinguishable.Write(s);
		fire_audio_weight.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/TorchComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class TorchComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    probability_of_ignition_attempt:NoitaInt = field(default_factory=NoitaInt)
    suffocation_check_offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    frames_suffocated_to_extinguish:NoitaInt = field(default_factory=NoitaInt)
    extinguishable:NoitaBool = field(default_factory=NoitaBool)
    fire_audio_weight:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.probability_of_ignition_attempt.Read(s)
        self.suffocation_check_offset_y.Read(s)
        self.frames_suffocated_to_extinguish.Read(s)
        self.extinguishable.Read(s)
        self.fire_audio_weight.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.probability_of_ignition_attempt.Write(s)
        self.suffocation_check_offset_y.Write(s)
        self.frames_suffocated_to_extinguish.Write(s)
        self.extinguishable.Write(s)
        self.fire_audio_weight.Write(s)

